### PR TITLE
[110][111][112][113] Add additional seat-related fields to Seat struct

### PR DESCRIPTION
**Pull request by shijithkjayan**
## Title

[Provide a succinct and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"]

## Type of Change

- [x] New feature

## Description

This PR expands the `Seat` struct in the `ExPass.Structs.SemanticTags.Seat` module by adding several optional fields:
- `seat_identifier`: A unique identifier for the seat (e.g., "Aisle 12, Row 3, Seat 5").
- `seat_number`: The seat number (e.g., "3A").
- `seat_row`: The row of the seat (e.g., "3").
- `seat_section`: The section of the seat (e.g., "Aisle 12").

#### Changes include:
- Updating the struct definition to include the new fields.
- Extending the `new/1` function to handle these fields with proper validation.
- Updating documentation and doctests to reflect the new fields.
- Adding comprehensive tests for each new field, covering creation, validation, whitespace trimming, JSON encoding, and handling of empty strings.

This addresses the following issues:
- #110 
- #111 
- #112 
- #113 

## Testing

Added extensive unit tests covering:
- Creation of Seat struct with and without each new field.
- JSON encoding verification for each new field.
- Whitespace trimming functionality for each new field.
- Validation of non-string inputs, ensuring appropriate errors are raised.
- Handling of empty strings for each new field.

All tests pass successfully.

## Impact

- Enhances the descriptive capabilities of the Seat struct.
- No breaking changes; all new fields are optional.
- Minimal performance impact due to lightweight validation and trimming operations.

## Additional Information

None

## Checklist

- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings

### CODE REVIEW

File: test/structs/semantic_tags/seat_test.exs [hunk 1]
```diff
@@ -94,4 +94,172 @@ defmodule ExPass.Structs.SemanticTags.SeatTest do
...
+
+    test "returns error for non-string seat_section" do
+      params = %{seat_section: 123}
+
+      assert_raise ArgumentError, "seat_section must be a string if provided", fn ->
+        Seat.new(params)
+      end
+    end
+
+    test "allows empty string for seat_section" do
```
Comment (njausteve) [inline test/structs/semantic_tags/seat_test.exs:258]: **question:** 

should it?

Comment (njausteve) [reply test/structs/semantic_tags/seat_test.exs:258]: I think for now, this is good for this PR. We can have a separate task that updates all string to ensure we don't allow empty strings.

**Review by njausteve — APPROVED** [review]
LGTM
